bfdd/control.c

```diff
@@ -774,13 +774,13 @@ static void _control_notify(struct bfd_control_socket *bcs,
 	control_queue_enqueue(bcs, bcm);
 }
 
-int control_notify(struct bfd_session *bs)
+int control_notify(struct bfd_session *bs, uint8_t notify_state)
```

Comment:
Please, don't change this function signature. All calls of this function you are already using what is inside `struct bfd_session`.

Copied from: https://github.com/FRRouting/frr/pull/5002#discussion_r325663118

Reply:
Refer File bfd.c: Line 353, below code snippet.

Peer state is being passed which is not available in struct bfd_sesison.

if (old_state == PTM_BFD_UP) {
		if (peer_state == PTM_BFD_ADM_DOWN)
			notify_state = peer_state;

		control_notify(bfd, notify_state);
	}

Reply:
Correct, we can fix that at comment https://github.com/FRRouting/frr/pull/5017?_pjax=%23js-repo-pjax-container#discussion_r327089050 .